frontend/server/src/DAO/Contests.php

```diff
@@ -634,6 +634,113 @@ final public static function getContestsParticipating(
         ];
     }
 
+    /**
+     * Returns the next contest a user registered for.
+     *
+     * @return ContestListItem
+     */
+    final public static function getNextRegisteredContestForUser(
+        int $identityId
+    ) {
+        $activeCondition = \OmegaUp\DAO\Enum\ActiveStatus::sql(
+            \OmegaUp\DAO\Enum\ActiveStatus::ACTIVE
+        );
+        $futureCondition = \OmegaUp\DAO\Enum\ActiveStatus::sql(
+            \OmegaUp\DAO\Enum\ActiveStatus::FUTURE
+        );
+        $columns = \OmegaUp\DAO\Contests::$getContestsColumns;
+
+        $select = "SELECT
+                        $columns,
+                        p.scoreboard_url,
+                        p.scoreboard_url_admin,
+                        COUNT(contestants.identity_id) AS contestants,
+                        ANY_VALUE(organizer.username) AS organizer";
+
+        $sql = "
+            FROM
+                (SELECT
+                    pi.problemset_id
+                FROM
+                    Problemset_Identities pi
+                WHERE
+                    pi.identity_id = ?
+                UNION DISTINCT
+                SELECT
+                    p.problemset_id
+                FROM
+                    Groups_Identities gi
+                INNER JOIN
+                    Group_Roles gr ON gi.group_id = gr.group_id
+                INNER JOIN
+                    Problemsets p ON gr.acl_id = p.acl_id
+                WHERE
+                    gi.identity_id = ? AND gr.role_id = ?
+                UNION DISTINCT
+                SELECT
+                    p.problemset_id
+                FROM
+                    Teams t
+                INNER JOIN
+                    Teams_Group_Roles tgr ON t.team_group_id = tgr.team_group_id
+                INNER JOIN
+                    Problemsets p ON tgr.acl_id = p.acl_id
+                WHERE
+                    t.identity_id = ? AND tgr.role_id = ?
+                ) pps
+            INNER JOIN
+                Contests
+            ON
+                Contests.problemset_id = pps.problemset_id
+            INNER JOIN
+                Problemsets p
+            ON
+                p.problemset_id = Contests.problemset_id
+            LEFT JOIN
+                Problemset_Identities AS contestants
+            ON
+                Contests.problemset_id = contestants.problemset_id
+            INNER JOIN
+                ACLs AS a
+            ON
+                Contests.acl_id = a.acl_id
+            INNER JOIN
+                Identities AS organizer
+            ON
+                a.owner_id = organizer.user_id
+            WHERE
+                ($activeCondition OR
+                $futureCondition) AND
+                archived = 0
+            GROUP BY Contests.contest_id
+        ";
+        $params = [
+            // Direct participation
+            $identityId,
```

Comment:
Acá cambiaría el parámetro por:
```suggestion
            $identity->identity_id,
```
Igual en los siguientes lugares donde usabas `$identityId`